components/range_cache_memory_engine/src/engine.rs

```diff
@@ -349,6 +349,14 @@ impl RangeCacheMemoryEngine {
     ) -> RangeCacheStatus {
         let mut core = self.core.write();
         let range_manager = core.mut_range_manager();
+        if range_manager.overlap_with_preferred_range(&region) {
```

Comment:
We should avoid doing this kind of heavy check in this hot path.

I suggest check preferred_ranges in following condition:
1. When preferred range changes, trigges a batch load for all regions(leader) fall in this range
2. When region becomes leader, check and may load target region.